src/app/pages/viewing/viewing.service.ts

```diff
@@ -0,0 +1,15 @@
+import { Injectable } from '@angular/core';
+import { MsgIoService } from '../../core/msg-io.service';
+import { VirtrolioMessage } from '../../shared/interfaces';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ViewingService {
+  messages: VirtrolioMessage[];
+  isCardView = false;
+  constructor(private msgIo: MsgIoService) {
+    // Subscribe to the array of VirtrolioMessage observables and make data available in messages variable
```

Comment:
I think this comment is unnecessary as the code variables are well named. I think some succinct documentation on top of the constructor would be more beneficial.

`/**
* Retrieve message data 
*/`